hbase-common/src/test/java/org/apache/hadoop/hbase/util/TestStrings.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.util;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class TestStrings {
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testAppendKeyValue() {
+        Assert.assertEquals(
+                new StringBuilder("foo, bar=baz"), Strings.appendKeyValue(
+                        new StringBuilder("foo"), "bar", "baz"));
+        Assert.assertEquals(
+                new StringBuilder("bar->baz"), Strings.appendKeyValue(
+                        new StringBuilder(), "bar", "baz", "->", "| "));
+        Assert.assertEquals(
+                new StringBuilder("foo, bar=baz"), Strings.appendKeyValue(
+                        new StringBuilder("foo"), "bar", "baz", "=", ", "));
+        Assert.assertEquals(
+                new StringBuilder("foo| bar->baz"), Strings.appendKeyValue(
+                        new StringBuilder("foo"), "bar", "baz", "->", "| "));
+    }
+
+    @Test
+    public void testDomainNamePointerToHostName() {
```

Comment:
The domainNamePointerToHostName(String) method is removing the trailing `.` from the input string. You did not cover this scenario in the test cases. Could you create one?

Reply:
Sorry I seemed to have forgotten to add that test